src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1436,6 +1437,34 @@ public function log(CompilerPassInterface $pass, string $message)
         $this->getCompiler()->log($pass, $this->resolveEnvPlaceholders($message));
     }
 
+    /**
+     * Checks whether a class is available and will remain available in the "no-dev" mode of Composer.
+     *
+     * When parent packages are provided and if any of them is in dev-only mode,
+     * the class will be considered available even if it is also in dev-only mode.
+     */
+    final public static function willBeAvailable(string $package, string $class, array $parentPackages): bool
+    {
+        if (!class_exists($class) && !interface_exists($class, false) && !trait_exists($class, false)) {
+            return false;
+        }
+
+        if (!class_exists(InstalledVersions::class) || !InstalledVersions::isInstalled($package) || InstalledVersions::isInstalled($package, true)) {
+            return true;
+        }
+
+        // the package is installed but in dev-mode only, check if this applies to one of the parent packages too
+
+        $rootPackage = InstalledVersions::getRootPackage()['name'] ?? '';
+        foreach ($parentPackages as $package) {
```

Comment:
I would avoid overriding the `$package` variable passed as argument here. This will reduce confusion when reading the code (and reduce the risk of bugs if we later add more code after this loop expecting to use the argument). I suggest using `foreach ($parentPackages as $parentPackage) {`

Reply:
updated